test/asserts.go

```diff
@@ -175,15 +175,16 @@ func AssertByteEquals(t *testing.T, one []byte, two []byte) {
 	}
 }
 
-// AssertContains determines whether needle can be found in haystack
+// AssertContains determines whether needle can be found in haystack.
+// A nil check on haystack should be performed prior to calling AssertContains.
```

Comment:
haystack is a string, it cannot be nil. It can be the empty string, but that shouldn't result in a panic, it should just result in `strings.Contains()` returning `false`.

Reply:
The input parameter populating haystack in my failure case from https://github.com/letsencrypt/boulder/issues/6748 is `err.Error()` which returns a string to satisfy haystack. The full line being:
```
test.AssertContains(t, err.Error(), "already revoked")
```
However,  `Error()` is a method of the `error` interface in the go stdlib errors package. This playground code does essentially the same thing and causes a panic. https://go.dev/play/p/AyxxV4etzoJ

Reply:
I believe that what's being suggested by this comment isn't possible; you cannot perform a nil check inside or outside of this function on specifically `haystack` as it's a string. This type of runtime panic is usually averted by checking if an `err != nil` before you do anything with it.

Reply:
Yes, the issue isn't that `err.Error()` is nil, the issue is that `err` is nil, and therefore `.Error()` cannot be called on it.